### PR TITLE
:sparkles: Add Level Select screen and logic along with Resetting Level Logic

### PR DESCRIPTION
**Pull request by viclzhu**
Add 'LevelSelect.hs' which handles level select screen ui and logic
- Reset Level logic here  

Added new type 'Level' :: (Int, Environment) in 'Sokoban.hs'.
Restructure event handling in 'Game.hs' to use 'Level' instead of 'Environment' for resetting level logic

Compiles and runs properly (no automated tests run)

### CODE REVIEW

File: src/Game.hs [hunk 1]
```diff
@@ -32,14 +33,21 @@ app = App
...
--- Esc key to quit
-handleEvent :: Environment -> BrickEvent Name e -> EventM Name (Next Environment)
-handleEvent env (VtyEvent (V.EvKey V.KEsc [])) = halt env
-handleEvent env (VtyEvent (V.EvKey V.KUp [])) = continue $ move UpMv env
-handleEvent env (VtyEvent (V.EvKey V.KRight [])) = continue $ move RightMv env
-handleEvent env (VtyEvent (V.EvKey V.KDown [])) = continue $ move DownMv env
-handleEvent env (VtyEvent (V.EvKey V.KLeft [])) = continue $ move LeftMv env
-handleEvent env _                              = continue env
+-- Apply a function to a level's environment
+applyMoveLevel f lvl = (fst lvl, f (snd lvl))
```
Comment (and-le) [inline src/Game.hs]: Add annotation for function type

File: src/Sokoban.hs [hunk 2]
```diff
@@ -36,6 +37,9 @@ data Cell = MkCell {
 }
     deriving (Eq, Show)
 
+-- Level number and its environment
+type Level = (Int, Environment)
```
Comment (and-le) [inline src/Sokoban.hs]: Do you anticipate needing to associate any other data with a level? If so, maybe a new `data` type class may be easier to work with than the `type` synonym.

**Review by and-le — COMMENTED** [review]
Overall the changes look good! Please see the comments above. I'm able to use the level select to pick levels 0 - 2. This can be merged after #5.

File: src/Sokoban.hs (hunk 2 above)
Comment (viclzhu) [reply src/Sokoban.hs]: Not sure, but yeah, I refactored the code to use `data` instead.

File: src/Game.hs (hunk 1 above)
Comment (viclzhu) [reply src/Game.hs]: ACK

**Review by radioactivebean0 — APPROVED** [review]
Good starting point for level select. Similar comments to the review on #5. There are a lot of duplicate and unnecessary exports/imports that can be deleted later on in the development process.

**Comment by and-le** [comment]
Please fix merge conflicts before merging (I think it is only import conflicts; nothing major).

**Review by and-le — COMMENTED** [review]
The updates for the different UI screens and level select logic look good! Once the merge conflicts are resolved, please merge this branch.

**Pull request merged**